lib/utils.js

```diff
@@ -0,0 +1,67 @@
+"use strict";
+
+function resolveHttpError(response, message) {
+    if (response.statusCode < 400) {
+        return;
+    }
+
+    message = message
+        ? message + " "
+        : "";
+
+    message += "Status code " + response.statusCode + " (" + response.statusMessage + ").";
+
+    if (response.body) {
+        message += " " + response.body;
+    }
+
+    var error = new Error(message);
+    error.response = response;
+
+    return error;
+}
+
+function createCallback(action, cb) {
+    return function (err, response) {
+        err = err || resolveHttpError(response, "Error " + action + ".");
+
+        if (err) {
+            return cb(err);
+        }
+
+        cb(null, response);
+    };
+}
+
+function createBodyCallback(action, cb) {
+    return createCallback(action, function (err, response) {
+        if (err) {
+            return cb(err);
+        }
+
+        cb(null, response.body, response);
+    });
+}
+
+function createJsonCallback(action, cb) {
+    return createCallback(action, function (err, response) {
+        if (err) {
+            return cb(err);
+        }
+
+        try {
+            var data = JSON.parse(response.body);
```

Comment:
That sounds reasonable. Does `json: true` gracefully handle scenarios where the response isn't json?

e.g. `kudu.vfs.getFile` returns the raw file contents as a string.

Even if we can't eliminate one of the callback functions, it could be nice to let `request` do the JSON parsing when we expect a JSON response.